onnxruntime/core/providers/xnnpack/math/gemm.cc

```diff
@@ -0,0 +1,194 @@
+                                                                                                         : 1;
+    } else {
+      K = A.Shape()->dim_size() == 3 ? A.Shape()->dim(2).dim_value() : A.Shape()->dim(1).dim_value();
+      M = A.Shape()->dim_size() == 3 ? A.Shape()->dim(1).dim_value() : A.Shape()->dim(0).dim_value() > 1 ? A.Shape()->dim(0).dim_value()                                                                                                 : 1;
+    }
+
+    if (trans_B_ == CblasNoTrans) {
+      N = B.Shape()->dim_size() == 3 ? B.Shape()->dim(2).dim_value() : B.Shape()->dim(1).dim_value();
+    } else {
+      N = B.Shape()->dim_size() == 3 ? B.Shape()->dim(1).dim_value() : B.Shape()->dim(0).dim_value() > 1 ? B.Shape()->dim(0).dim_value()                                                                                                  : 1;
+    }
+  }
+
+  Status Gemm::PrePack(const Tensor& tensor, int input_idx, AllocatorPtr alloc,
+                       /*out*/ bool& is_packed,
+                       /*out*/ PrePackedWeights*) {
+    is_packed = false;
+
+    if (input_idx == 0) {
+      return Status::OK();
+    }
+
+    if (input_idx == 1) {
+      B_ = Tensor(tensor.DataType(), TensorShape(tensor.Shape()), alloc);
+      SingleAxisTranspose(std::vector<size_t>{0, 1}, tensor, B_, /*from*/ 1, /*to*/ 1);
+
+      return Status::OK();
+    }
+
+    is_packed = true;
+
+    uint32_t flags = trans_B_ != CblasTrans ? XNN_FLAG_TRANSPOSE_WEIGHTS:0;
+
+    float output_min = clip_min_max_ ? clip_min_max_->first : -INFINITY;
+    float output_max = clip_min_max_ ? clip_min_max_->second : INFINITY;
+
+    if (input_idx == 2) {
+      xnn_status status = xnn_status::xnn_status_uninitialized;
+      struct xnn_operator* p = nullptr;
+      status = xnn_create_fully_connected_nc_f32(
+          B_.Shape()[0],         // size_t input_channels,
+          B_.Shape()[1],         // size_t output_channels,
+          B_.Shape()[0],         // size_t input_stride,
+          B_.Shape()[1],         // size_t output_stride,
+          B_.Data<float>(),      // const float* kernel,
+          tensor.Data<float>(),  // const float* bias,
+          output_min,
+          output_max,
+          flags,
+#ifdef XNN_CACHE_ENABLE
+          &xnn_caches_,
+#else
+          0,
+#endif
+          &p);
+
+      if (status != xnn_status_success) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "xnn_create_fully_connected_nc_f32 returned ", status);
+      }
+      op0_.reset(p);
+    }
+
+    return Status::OK();
+  }
+
+  Status Gemm::Compute(OpKernelContext* context) const {
+    pthreadpool_t t_pool = GetThreadPool();
+    const auto* A = context->Input<Tensor>(0);
+    auto Y = context->Output(0, {M, N});
+
+    // if input is empty tensor, return as nothing need to be calculated and we've set the shape for the output
+    if (M == 0 || N == 0)
+      return Status::OK();
+    
+    xnn_status status = xnn_setup_fully_connected_nc_f32(
```

Comment:
we should check it also, otherwise it will be shadowed by the nest `status`

Reply:
Correct - I fixed that. Thanks for highlighting this